pslab/external/MLX90614.py

```diff
@@ -1,46 +1,40 @@
-from __future__ import print_function
+from pslab.bus import I2CSlave
 
-
-def connect(route, **args):
-    return MLX90614(route, **args)
-
-
-class MLX90614():
+class MLX90614(I2CSlave):
+    _ADDRESS = 0x5A
+    _OBJADDR = 0x07
+    _AMBADDR = 0x06
     NUMPLOTS = 1
     PLOTNAMES = ['Temp']
-    ADDRESS = 0x5A
     name = 'PIR temperature'
 
-    def __init__(self, I2C, **args):
-        self.I2C = I2C
-        self.ADDRESS = args.get('address', self.ADDRESS)
-        self.OBJADDR = 0x07
-        self.AMBADDR = 0x06
+    def __init__(self):
+        super().__init__(self._ADDRESS)
 
-        self.source = self.OBJADDR
+        self.source = self._OBJADDR
 
         self.name = 'Passive IR temperature sensor'
         self.params = {'readReg': {'dataType': 'integer', 'min': 0, 'max': 0x20, 'prefix': 'Addr: '},
                        'select_source': ['object temperature', 'ambient temperature']}
 
-        try:
-            print('switching baud to 100k')
-            self.I2C.configI2C(100e3)
-        except Exception as e:
-            print('FAILED TO CHANGE BAUD RATE', e.message)
+        # try:
+        #     print('switching baud to 100k')
+        #     self.I2C.configI2C(100e3)
+        # except Exception as e:
+        #     print('FAILED TO CHANGE BAUD RATE', e.message)
 
     def select_source(self, source):
         if source == 'object temperature':
-            self.source = self.OBJADDR
+            self.source = self._OBJADDR
         elif source == 'ambient temperature':
-            self.source = self.AMBADDR
+            self.source = self._AMBADDR
 
     def readReg(self, addr):
         x = self.getVals(addr, 2)
         print(hex(addr), hex(x[0] | (x[1] << 8)))
 
     def getVals(self, addr, numbytes):
-        vals = self.I2C.readBulk(self.ADDRESS, addr, numbytes)
+        vals = self.read(numbytes, addr)
         return vals
 
     def getRaw(self):
```

Comment:
_private, snake_case.

The method itself could be clearer.

- It fetches three bytes from the sensor, but only uses two of them. Why?
- This: `[((((vals[1] & 0x007f) << 8) + vals[0]) * 0.02) - 0.01 - 273.15]` is some kind of conversion, but what exactly is going on?
    - The most significant bit of `val[1]` is discarded, why?
    - 0.02 is a resolution factor (0.02 degrees of resolution per bit), but what is 0.01?
    - 273.15 is conversion from degK to degC.

The method returns False if fewer than three bytes are received from the sensor. Can this happen in normal operation, or is it indicative of a hardware problem? If the latter, we should raise an exception instead.

The return type can be either list or bool. Multiple return types should be avoided; better return an empty list instead of False.